### PR TITLE
Migrate standalone widgets editor to use new preferences package

### PR DESCRIPTION
**Pull request by talldan**
## Description
Addresses some of #31965, follows #38873

Migrates the standalone widgets editor to use the new preferences package (introduced in #38873). Previously it was using `interface`, but the plan is now to deprecate the preferences APIs in interface in favor of this new package.

The main goal here is that nothing is different from a user perspective.

## Testing Instructions
1. Delete your WP localstorage data
2. Using `trunk` open the widgets editor and dismiss the welcome guide
3. Toggle some of the preferences in the More Menu to something other than the default (e.g. set Top toolbar to active)
4. Checkout this branch and rebuild
5. Reload the widgets editor
6. The welcome guide should not show and the preferences set in step 3 should still be the same (the data migration worked)
7. Change a few more preferences
8. Reload
9. The preferences should still be the same from step 7 (preference changes are still persisted)

## Types of changes
What types of changes does your code introduce?

## Checklist:
- [x] My code is tested.
- [x] My code follows the WordPress code style. 
- [x] My code follows the accessibility standards. 
- [x] I've tested my changes with keyboard and screen readers. 
- [x] My code has proper inline documentation. 
- [x] I've included developer documentation if appropriate. 
- [x] I've updated all React Native files affected by any refactorings/renamings in this PR (please manually search all `*.native.js` files for terms that need renaming or removal). 
- [x] I've updated related schemas if appropriate.

### CODE REVIEW

File: packages/base-styles/_z-index.scss [hunk 1]
```diff
@@ -155,8 +155,7 @@ $z-layers: (
 	".components-popover.customize-widgets-more-menu__content": 99998,
 	".components-popover.edit-post-more-menu__content": 99998,
 	".components-popover.edit-site-more-menu__content": 99998,
-	".components-popover.edit-widgets-more-menu__content": 99998,
-	".components-popover.preferences-more-menu__content": 99998,
```
Comment (talldan) [inline packages/base-styles/_z-index.scss:159]: Seems like I accidentally added this in https://github.com/WordPress/gutenberg/pull/38873. The more menu is staying in the interface package, so the 'preferences' part here is wrong.

It shouldn't have caused any bugs as nothing should be using this name. As I migrate each editor, the individual z-indices for each editor can be removed in favor of just the `interface` one.

File: packages/interface/src/components/more-menu-dropdown/style.scss [hunk 2]
```diff
@@ -31,5 +31,5 @@
 }
 
 .components-popover.interface-more-menu-dropdown__content {
-	z-index: z-index(".components-popover.edit-widgets-more-menu__content");
+	z-index: z-index(".components-popover.interface-more-menu__content");
```
Comment (talldan) [inline packages/interface/src/components/more-menu-dropdown/style.scss:34]: This was wrong for some reason, so I'm changing it here to match the change in the _z-index.scss file.

**Review by draganescu — APPROVED** [review]
Excellent stuff. Tested works as described and the code is very straightforward.

**Review by youknowriad — COMMENTED** [review]
Can we now update the widgets store to use `register` instead of `registerStore` ?

**Comment by talldan** [comment]
> Can we now update the widgets store to use register instead of registerStore ?

We can, though it looks like it didn't need to use `registerStore` anyway. I've updated it.

**Comment by noisysocks** [comment]
When I `nvm use; npm install -g npm@6; rm -rf node_modules; npm install`, I see local changes to `package-lock.json`. But I'm not convinced it's true because wouldn't tests fail if that were the case? 🙃

Anyway, testing this and looking at the code now 😀

**Comment by talldan** [comment]
> When I nvm use; npm install -g npm@6; rm -rf node_modules; npm install, I see local changes to package-lock.json. But I'm not convinced it's true because wouldn't tests fail if that were the case? 🙃

Yeah, they would fail, I already had to wrestle them once or twice. What changes are you seeing?

File: packages/edit-widgets/src/store/index.js [hunk 3]
```diff
@@ -35,10 +35,7 @@ const storeConfig = {
  * @type {Object}
  */
 export const store = createReduxStore( STORE_NAME, storeConfig );
-
-// Once we build a more generic persistence plugin that works across types of stores
-// we'd be able to replace this with a register call.
-registerStore( STORE_NAME, storeConfig );
```
Comment (noisysocks) [inline packages/edit-widgets/src/store/index.js:41]: Just for my own clarity: Is there something in this PR that lets us remove this that I'm not understanding? Or is it a "while we're here" change?

**Review by noisysocks — APPROVED** [review]
Code looks good and this tests well! I like the unit tests included—convinces me that the migration code is good.

:shipit:

**Comment by noisysocks** [comment]
I did yet another `rm -rf node_modules; npm i` and now there are no changes 🤷‍♂️ Mondays...

File: packages/edit-widgets/src/store/index.js (hunk 3 above)
Comment (talldan) [reply packages/edit-widgets/src/store/index.js:41]: It's a 'while we're here' change. It should have been done previously when the edit-widgets store stopped using the persistence plugin (in https://github.com/WordPress/gutenberg/pull/33774).

Thanks for reviewing.

**Comment by talldan** [comment]
Next one up is the customize widget editor, which is addressed in https://github.com/WordPress/gutenberg/pull/39112.

File: packages/edit-widgets/src/store/index.js (hunk 3 above)
Comment (noisysocks) [reply packages/edit-widgets/src/store/index.js:41]: Ahhh gotcha.

**Pull request merged**